job_executor/domain/rollback.py

```diff
@@ -113,3 +114,58 @@ def rollback_bump(job_id: str, bump_manifesto: dict):
     except Exception as e:
         logger.error(f'{job_id}: Unexpected error when rolling back job')
         logger.exception(e)
+
+
+def rollback_worker_job(job_id: str, operation: str, dataset_name: str):
+    logger.info(
+        f'{job_id}: Rolling back worker job '
+        f'with target: "{dataset_name}" and operation "{operation}"'
+    )
+    generated_metadata_files = [
+        f'{dataset_name}.json',
+        f'{dataset_name}__DRAFT.json'
+    ]
+    generated_data_files = [
+        f'{dataset_name}.db',
+        f'{dataset_name}.csv',
+        f'{dataset_name}_pseudonymized.csv',
+        f'{dataset_name}_pseudonymized_enriched.csv',
+        f'{dataset_name}__DRAFT.parquet'
+    ]
+    generated_data_directory = f'{dataset_name}__DRAFT'
+
+    for file in generated_metadata_files:
+        filepath = WORKING_DIR_PATH / file
+        if filepath.exists:
+            logger.info(f'{job_id}: Deleting metadata file "{filepath}"')
+            os.remove(filepath)
+
+    if operation in ['ADD', 'CHANGE_DATA']:
+        for file in generated_data_files:
+            filepath = WORKING_DIR_PATH / file
+            if filepath.exists:
```

Comment:
It might be that the worker got interrupted before it managed to generate the: 
`f'{dataset_name}_pseudonymized_enriched.csv',` file, and in that case it does not exist.

The linter does not validate the contents of the filesystem, but there might be a point I'm missing here?

Reply:
This one needs a parenthesis too:  parquet_directory.exists  (line 139) :P